sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/_tests_/ece-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,574 @@
+import { DeepMocked, createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+  formatDate,
+} from "../../../../../../test/helpers";
+import { ECEResponseIntegrationScheduler } from "../ece-response-integration.scheduler";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  mockDownloadFiles,
+  saveFakeApplicationDisbursements,
+} from "@sims/test-utils";
+import * as Client from "ssh2-sftp-client";
+import { Job } from "bull";
+import * as path from "path";
+
+import { InstitutionLocationService } from "@sims/integrations/services";
+import { ECE_RESPONSE_FILE_NAME } from "@sims/integrations/constants";
+import { ProcessSummaryResult } from "@sims/integrations/models";
+import { ApplicationStatus } from "@sims/sims-db";
+
+describe(
+  describeProcessorRootTest(QueueNames.ECEProcessResponseIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: ECEResponseIntegrationScheduler;
+    let db: E2EDataSources;
+    let sftpClientMock: DeepMocked<Client>;
+    let eceResponseMockDownloadFolder: string;
+    let institutionLocationService: InstitutionLocationService;
+
+    beforeAll(async () => {
+      eceResponseMockDownloadFolder = path.join(
+        __dirname,
+        "ece-response-files",
+      );
+      // Set the institution response folder to the mock folder.
+      process.env.INSTITUTION_RESPONSE_FOLDER = eceResponseMockDownloadFolder;
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      // Processor to be tested.
+      processor = app.get(ECEResponseIntegrationScheduler);
+      institutionLocationService = app.get(InstitutionLocationService);
+    });
+
+    it("Should process an ECE response file and confirm the enrolment when the disbursement and application is valid.", async () => {
+      // Arrange
+      // Mock institution code to confirm the enrolment.
+      const confirmEnrolmentInstitutionCode = "CONF";
+      const confirmEnrolmentResponseFile = path.join(
+        process.env.INSTITUTION_RESPONSE_FOLDER,
+        confirmEnrolmentInstitutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      );
+      // Institution location service mock to return mocked institution code.
+      institutionLocationService.getAllIntegrationEnabledInstitutionCodes =
```

Comment:
Even though the method `getAllIntegrationEnabledInstitutionCodes` is simple, the idea of the E2E tests would be to also test the DB queries. Thoughts?